mesonpy/_editable.py

```diff
@@ -304,12 +304,21 @@ def rebuild(self) -> Node:
         env[MARKER] = os.pathsep.join((env.get(MARKER, ''), self._build_path))
 
         if self._verbose or bool(env.get(VERBOSE, '')):
-            print('+ ' + ' '.join(self._build_cmd))
-            stdout = None
+            # We do not want any output if there is no work to do. Code is adapted from:
+            # https://github.com/mesonbuild/meson/blob/a35d4d368a21f4b70afa3195da4d6292a649cb4c/mesonbuild/mtest.py#L1635-L1636
+            dry_run_build_cmd = self._build_cmd + ['-n']
+            p = subprocess.run(dry_run_build_cmd, cwd=self._build_path, env=env, capture_output=True)
+            if b'ninja: no work to do.' not in p.stdout and b'samu: nothing to do' not in p.stdout:
+                module_names = ' '.join(sorted(self._top_level_modules))
+                build_command = ' '.join(self._build_cmd)
+                info_msg = (
+                    f'+ meson-python: building {module_names}\n'
+                    f'+ meson-python build command: {build_command}'
```

Comment:
For consistency, I would spell this `meson-python: executing {build_command}`.

Reply:
Or you can make it a single line, combining the two messages: `meson-python: building {module_names}: {build_command}`. And I would loose the `+ ` prefix.